File: Kernal64/src/ucesoft/cbm/peripheral/sid/SIDChip.scala
package ucesoft.cbm.peripheral.sid

import java.io.{ObjectInputStream, ObjectOutputStream}

trait SIDChip {
  def clock : Unit
  def read(offset : Int) : Int
  def write(offset : Int,value : Int) : Unit
  def setModel(mode : Int) : Unit
  def output : Int

  def saveState(out:ObjectOutputStream) : Unit
  def loadState(in:ObjectInputStream) : Unit

  def reset : Unit

  def updateBusValue(value : Int) : Unit
}

File: Kernal64/src/ucesoft/cbm/peripheral/sid/SIDChip.java
package ucesoft.cbm.peripheral.sid;

import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

public interface SIDChip {
    void clock();
    int read(int offset);
    void write(int offset,int value);
    void setModel(int model);
    int output();

    void saveState(ObjectOutputStream out);
    void loadState(ObjectInputStream in);

    void reset();
    void updateBusValue(int value);
}
